ambry-clustermap/src/main/java/com/github/ambry/clustermap/DataNodeConfig.java

```diff
@@ -0,0 +1,229 @@
+import java.util.Map;
+import java.util.Set;
+
+
+/**
+ * A data object for configs scoped to a single data node.
+ */
+class DataNodeConfig {
+  private final String instanceName;
+  private final String hostName;
+  private final int port;
+  private final String datacenterName;
+  private final Integer sslPort;
+  private final Integer http2Port;
+  private final String rackId;
+  private final long xid;
+  private final Set<String> sealedReplicas = new HashSet<>();
+  private final Set<String> stoppedReplicas = new HashSet<>();
+  private final Map<String, DiskConfig> diskConfigs = new HashMap<>();
+
+  /**
+   * @param instanceName a name that can be used as a unique key for this server.
+   * @param hostName the host name of the server.
+   * @param port the port of the server.
+   * @param datacenterName the datacenter this server is in.
+   * @param sslPort the ssl port, or {@code null} if the server does not have one.
+   * @param http2Port the HTTP2 port, or {@code null} if the server does not have one.
+   * @param rackId an identifier for the rack or cabinet that the server is in for computing failure domains.
+   * @param xid  the xid for this server.
+   */
+  DataNodeConfig(String instanceName, String hostName, int port, String datacenterName, Integer sslPort,
+      Integer http2Port, String rackId, long xid) {
+    this.instanceName = instanceName;
+    this.hostName = hostName;
+    this.port = port;
+    this.datacenterName = datacenterName;
+    this.sslPort = sslPort;
+    this.http2Port = http2Port;
+    this.rackId = rackId;
+    this.xid = xid;
+  }
+
+  /**
+    * @return a name that can be used as a unique key for this server.
+   */
+  String getInstanceName() {
+    return instanceName;
+  }
+
+  /**
+   * @return the host name of the server.
+   */
+  String getHostName() {
+    return hostName;
+  }
+
+  /**
+   * @return the port of the server.
+   */
+  int getPort() {
+    return port;
+  }
+
+  /**
+   * @return the datacenter this server is in.
+   */
+  String getDatacenterName() {
+    return datacenterName;
+  }
+
+  /**
+   * @return the ssl port, or {@code null} if the server does not have one.
+   */
+  Integer getSslPort() {
+    return sslPort;
+  }
+
+  /**
+   * @return the HTTP2 port, or {@code null} if the server does not have one.
+   */
+  Integer getHttp2Port() {
+    return http2Port;
+  }
+
+  /**
+   * @return an identifier for the rack or cabinet that the server is in for computing failure domains.
+   */
+  String getRackId() {
+    return rackId;
+  }
+
+  /**
+   * @return the xid for this server. After {@link SimpleClusterChangeHandler} is retired, this field will be removed.
+   */
+  long getXid() {
+    return xid;
+  }
+
+  /**
+   * @return the set of sealed replicas on this server. This set is mutable.
+   */
+  Set<String> getSealedReplicas() {
+    return sealedReplicas;
+  }
+
+  /**
+   * @return the set of stopped replicas on this server. This set is mutable.
```

Comment:
I wonder why these two sets are mutable?

Reply:
I have not implemented this in this PR, but I was planning on `HelixParticipant` directly modifying these sets in DataNodeConfig for things like adding replicas and sealing partitions. If you prefer, I can introduce a builder with a copy constructor, but I felt like that was not needed given that this is similar to how `InstanceConfig` is treated/mutated in `HelixParticipant`.